pkg/auth/rights/auth_info.go

```diff
@@ -47,3 +48,22 @@ func AuthInfo(ctx context.Context) (authInfo *ttnpb.AuthInfoResponse, err error)
 	}
 	return authInfo, nil
 }
+
+var errUnauthenticated = errors.DefineUnauthenticated("unauthenticated", "unauthenticated")
+
+// RequireAuthentication confirms if the authentication information within a context contains any rights, if so,
+// the user is considered to be authenticated.
+func RequireAuthentication(ctx context.Context) error {
+	log.FromContext(ctx).Debug("Validating request authentication")
+	authInfo, err := AuthInfo(ctx)
+	if err != nil {
+		log.FromContext(ctx).WithError(err).Debug("Failed to validate authentication information")
+		return err
+	}
+
+	if authInfo.GetAccessMethod() == nil && len(authInfo.GetUniversalRights().GetRights()) == 0 {
+		log.FromContext(ctx).Debug("User not authenticated")
```

Comment:
Is this really a user authentication? Wouldn't call authentication or request authentication be more generic?

Also since the error is being returned to the caller and there's no really ambiguity for the cause, I don't think this needs to be logged.

Reply:
Regarding the question of if this is really an user authentication, not sure if what you meant is similar to what I brought it up on the issue [discussion](https://github.com/TheThingsNetwork/lorawan-stack/issues/5512#issuecomment-1200517714). If so I indeed agree, the reason for not changing it in this PR was it that the validation seems to be sufficient for the DR and it will be changed to be a tad bit more generic when working on `Updating the rpcmetadata.WithForwardedAuth`.

Or perhaps you are talking about the rights validation? If so I'm not too sure what you meant

Reply:
See https://github.com/TheThingsNetwork/lorawan-stack/pull/5659#discussion_r936615531